File: Park-Android/app/src/main/java/com/dashang/park/helper/StringHelper.java
package com.dashang.park.helper;

import android.text.TextUtils;

import org.w3c.dom.Text;

import java.text.DecimalFormat;


public class StringHelper {

    public static final String ANDROID_ARM="http://zgpulai.com/park_arm.apk";
    public static final String ANDROID_X86="http://zgpulai.com/park_x86.apk";
    public static final String IOS="http://itunes.com/apps/";
    public static final String QQ_APP_ID = "101210291";
    public static final String WECHAT_APP_ID = "wxe83bdfef70a4ba4f";
    public static final String WECHAT_SECRET = "ebfd1fe4fd57b890d57fc4dd3ecc7e96";

    public static boolean isPasswordValid(String password) {
        return password.matches("(?!^[0-9]*$)(?!^[a-zA-Z]*$)^([a-zA-Z0-9]{6,20})$");
    }

    public static String getLoc(String loc) {
        String result;
        if(TextUtils.isEmpty(loc)){
            result = "未停车";
        }else {
            result = "郑州大商新玛特金博大店";
            if (loc.startsWith("L2")) {
                result += "负二层";
            } else if (loc.startsWith("L3")) {
                result += "负三层";
            }
            String[] parts = loc.split("-");
            if(parts.length>1) {
                result += " " + parts[1];
            }else{
                result = "未停车";
            }
        }
        return result;
    }


    public static boolean isEmailValid(String email) {
        String ePattern = "^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
        java.util.regex.Pattern p = java.util.regex.Pattern.compile(ePattern);
        java.util.regex.Matcher m = p.matcher(email);
        return m.matches();
    }


    public static boolean isPhoneValid(String phone) {
        if (phone.length() < 6) {
            return false;
        }
        String ePattern = "^[0-9]*$";
        java.util.regex.Pattern p = java.util.regex.Pattern.compile(ePattern);
        java.util.regex.Matcher m = p.matcher(phone);
        return m.matches();
    }

}

File: Park-iOS/StringHelper.swift
//
//  StringHelper.swift
//  Park-iOS
//
//  Created by Dieter Kohl on 5/14/15.
//  Copyright (c) 2015 Dieter Kohl. All rights reserved.
//

import Foundation

class stringHelper : NSObject{

    class var ANDROID_ARM : String { return "http://zgpulai.com/park_arm.apk" }
    
    class var ANDROID_X86 : String { return  "http://zgpulai.com/park_x86.apk"}
    class var IOS : String { return  "http://zgpulai.com/park_arm.apk"}

    class var QQ_APP_ID : String { return "101210291"}

    class var WECHAT_APP_ID : String { return  "wxe83bdfef70a4ba4f"}

    class var WECHAT_SECRET : String { return  "ebfd1fe4fd57b890d57fc4dd3ecc7e96"}
    
    class var BASE_URL : String{return "http://zgpulai.com/api/controller/"}
    
    class func isEmailValid(testStr:String) -> Bool {
        // println("validate calendar: \(testStr)")
        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}"
        
        let emailTest = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
        return emailTest.evaluateWithObject(testStr)
    }
    

     class func validate(value: String) -> Bool {

        let PHONE_REGEX = "^\\d{10}$"

        var phoneTest = NSPredicate(format: "SELF MATCHES %@", PHONE_REGEX)
        return phoneTest.evaluateWithObject(value)
    }
    
    class func isPasswordValid(value: String) -> Bool
    {
        let PW_REGEX = "(?!^[0-9]*$)(?!^[a-zA-Z]*$)^([a-zA-Z0-9]{6,20})$"
        
        var phoneTest = NSPredicate(format: "SELF MATCHES %@", PW_REGEX)
        return phoneTest.evaluateWithObject(value)
    
    }
    
    class func getLoc(loc : String) -> String {
        var  result : String
        
        if loc.isEmpty == true{
            result = "未停车"
        }
        else {
            result = "郑州大商新玛特金博大店"
            if loc.hasPrefix("L2"){
                result += "负二层"
              }else if loc.hasPrefix("L3"){
                 result += "负三层"
            }
            
            let strArray = loc.componentsSeparatedByString("-")
            if strArray.count > 1{
                result += " " + strArray[1]
            }
            else{
                result = "未停车"
            }
        }
        
        return result
    }
}
